drivers/include/ltc4150.h

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2018 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_ltc4150 LTC4150 coulomb counter
+ * @ingroup     drivers_sensors
+ * @brief       Driver for LTC4150 coulomb counter
+ * @{
+ *
+ * @file
+ * @brief       LTC4150 coulomb counter
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * # Wiring the LTC4150
+ * Please note that this driver works interrupt driven and does not clear the
+ * signal. Thus, the CLR and INT pins on the LTC4150 need to be connected
+ * (solder jumper SJ1 closed on the breakout board), so that the signal is
+ * automatically cleared.
+ *
+ * When using the breakout board, check whether your board requires the solder
+ * jumpers SJ2 and SJ3 to be closed (3.3V logic) or open (5V logic). Connect
+ * the VIO pin to the logic level, GND to ground, IN+ and IN- to the power
+ * supply and use OUT+ and OUT- to power your board.
+ *
+ * In the easiest case only the /INT pin needs to be connected to a GPIO,
+ * /SHDN and POL can be left floating. The GPIO /INT is connected to support
+ * for interrupts, /SHDN and POL (if connected) do not require interrupt
+ * support.
+ *
+ * In case a battery is used, the module `ltc4150_battery` should also be added
+ * and the POL pin connected to another GPIO. This allows to distinguish between
+ * charge drawn from the battery and charge transferred into the battery
+ * (used to load it).
+ *
+ * In case support to power off the LTC4150 is desired, the /SHDN pin needs to
+ * be connected to a third GPIO.
+ *
+ * # Things to keep in mind
+ * The LTC4150 creates pulses with a frequency depending on the current drawn.
+ * Thus, more interrupts need to be handled when more current is drawn, which
+ * in turn increases system load (and power consumption). The interrupt service
+ * routing is quite short, so this effect should hopefully be negligible.
+ */
+
+#ifndef LTC4150_H
+#define LTC4150_H
+
+#include <mutex.h>
+#include <periph/gpio.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Data structure storing the number of pulses for (dis-)charging
+ *
+ * In addition to the total number of pulses the accumulated pulses are stored
+ * for:
+ * - Every second in the last minute
+ * - The last minute
+ */
+typedef struct {
+    uint32_t total;         /**< Total number of pulses */
+    uint16_t sec[60];       /**< Running average over each second in the last minute */
```

Comment:
Actually the motivation wasn't to keep track of each second and rather to allow updating the last minute value in `O(1)`, so that the ISR is not too expensive.

I just checked how many ticks I expected in a second as worst case. Lets assume we estimate the maximum current to `0.1A`, then we would choose `0.05V / 0.1A = 0.5Ohm` as `R_sense`. If we now would have a current of `1A` (ten times higher than estimated maximum), then we would get `32.55*0.5 = 16.275` ticks per coulomb (or 16.275 ticks per second with a current of 1A). Hence, it seems pretty safe to assume at most 20 ticks per second will occur. Thus, to achieve O(1) updates for the last minute counter only one byte per every ten seconds could be used (instead 2 bytes per second), or 6 bytes instead of 120 bytes.

Also: The per second value seems to be of little value, as e.g. the MSBA2 gets 0-1 ticks per second without any other hardware attached. I think this is simple to few sample to be of any value at all.

Reply:
6 bytes instead of 120 sounds much better. Just tracking the counter would simplify that but I see that it's nice to have that functionality - especially for the provided saul implementation.